lib/code-renderer.js

```diff
@@ -0,0 +1,20 @@
+var React = require('react');
+var $ = React.createElement;
+
+var highlight = require('./highlight');
+
+module.exports = function codeRenderer(node, props, children) {
+  var lang = (node.info || '').trim() || null;
+  var className = 'language-' + lang;
+
+  var highlightedCode = highlight(lang, children[0]);
+
+  var codeElement = $('code', {
+    className: className,
+    dangerouslySetInnerHTML: {
```

Comment:
I'm not sure about this, if we just spat out a `<code>` and then `hljs.highlightBlock(ref);` there'd be no need to `dangerouslySetInnerHTML`.